src/network/networkd-manager.c

```diff
@@ -1444,6 +1445,16 @@ int manager_new(Manager **ret, sd_event *event) {
 
         (void) routing_policy_load_rules(m->state_file, &m->rules_saved);
 
+        /* Router lifetime can be set with netlink interface since kernel >= 4.5
+         * so for the supported kernel we dont need to expire routes in userspace */
+        {
+                Condition c = { .type = CONDITION_KERNEL_VERSION, .parameter = (char *) ">= 4.5" };
+                r = condition_test(&c);
+                if (r < 0)
+                        return r;
+                m->route_expiration_kernel = !!r;
+        }
+
```

Comment:
instead of initializing this here during initialization time, I think we should determine this on demand only, i.e. introduce a new call like this:

```c
int kernel_route_expiration_supported(void) {
        static int cached = -1;
        int r;

        if (cached < 0) {
                Condition c = { 
                        .type = CONDITION_KERNEL_VERSION, 
                        .parameter = (char *) ">= 4.5" 
                };
                r = condition_test(&c);
                if (r < 0)
                        return r;

                cached = r;
        }
        return cached;
}
```

This  call would cache the result internally and only run the check on the first invocation.